honeycomb-core/src/cmap/builder/grid/building_routines.rs

```diff
@@ -17,88 +17,93 @@ pub fn build_2d_grid<T: CoordsFloat>(
                         ),
                 );
-                let last_column = x_idx == n_square_x - 1;
-                let last_row = y_idx == n_square_y - 1;
-                if last_column {
-                    // that last column of 0-cell needs special treatment
-                    // bc there are no "horizontal" associated dart
-                    let vertex_id = map.vertex_id(base_dart + 1);
-                    map.insert_vertex(
-                        vertex_id,
-                        origin
-                            + Vector2(
-                                T::from(x_idx + 1).unwrap() * len_per_x,
-                                T::from(y_idx).unwrap() * len_per_y,
-                            ),
-                    );
-                }
-                if last_row {
-                    // same as the case on x
-                    let vertex_id = map.vertex_id(base_dart + 3);
-                    map.insert_vertex(
-                        vertex_id,
-                        origin
-                            + Vector2(
-                                T::from(x_idx).unwrap() * len_per_x,
-                                T::from(y_idx + 1).unwrap() * len_per_y,
-                            ),
-                    );
-                }
-                if last_row & last_column {
-                    // need to do the upper right corner
-                    let vertex_id = map.vertex_id(base_dart + 2);
-                    map.insert_vertex(
-                        vertex_id,
-                        origin
-                            + Vector2(
-                                T::from(x_idx + 1).unwrap() * len_per_x,
-                                T::from(y_idx + 1).unwrap() * len_per_y,
-                            ),
-                    );
-                }
+            }
+            if last_row {
+                // same as the case on x
+                let vertex_id = map.vertex_id(d4);
+                map.insert_vertex(
+                    vertex_id,
+                    origin
+                        + Vector2(
+                            T::from(x_idx).unwrap() * len_per_x,
+                            T::from(y_idx + 1).unwrap() * len_per_y,
+                        ),
+                );
+            }
+            if last_row & last_column {
+                // need to do the upper right corner
+                let vertex_id = map.vertex_id(d3);
+                map.insert_vertex(
+                    vertex_id,
+                    origin
+                        + Vector2(
+                            T::from(x_idx + 1).unwrap() * len_per_x,
+                            T::from(y_idx + 1).unwrap() * len_per_y,
+                        ),
+                );
             }
         });
-    });
 
     // and then build faces
-    assert_eq!(map.fetch_faces().identifiers.len(), n_square_x * n_square_y);
+    debug_assert_eq!(map.fetch_faces().identifiers.len(), n_square_x * n_square_y);
```

Comment:
Leave a comment stating why it can impact the performances.

Reply:
done in 5d61aa9